wallet/wallet-node-client/src/rpc_client/cold_wallet_client.rs

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use chainstate::ChainInfo;
+use common::{
+    chain::{
+        tokens::{RPCTokenInfo, TokenId},
+        Block, DelegationId, GenBlock, PoolId, SignedTransaction, Transaction,
+    },
+    primitives::{Amount, BlockHeight, Id},
+};
+use consensus::GenerateBlockInputData;
+use crypto::ephemeral_e2e::EndToEndPublicKey;
+use mempool::{tx_accumulator::PackingStrategy, tx_options::TxOptionsOverrides, FeeRate};
+use p2p::{
+    interface::types::ConnectedPeer,
+    types::{bannable_address::BannableAddress, ip_or_socket_address::IpOrSocketAddress, PeerId},
+};
+
+use crate::node_traits::NodeInterface;
+
+use super::ColdWalletClient;
+
+#[derive(thiserror::Error, Debug)]
+pub enum ColdWalletRpcError {
+    #[error("Method is not available in cold wallet mode")]
+    NotAvailable,
+}
+
+#[async_trait::async_trait]
+impl NodeInterface for ColdWalletClient {
+    type Error = ColdWalletRpcError;
+
+    async fn chainstate_info(&self) -> Result<ChainInfo, Self::Error> {
+        let genesis = self.chain_config.genesis_block();
+        Ok(ChainInfo {
+            best_block_id: self.chain_config.genesis_block_id(),
+            best_block_height: BlockHeight::zero(),
+            best_block_timestamp: genesis.timestamp(),
+            median_time: genesis.timestamp(),
+            is_initial_block_download: false,
+        })
+    }
+
+    async fn get_block(&self, _block_id: Id<Block>) -> Result<Option<Block>, Self::Error> {
+        Err(ColdWalletRpcError::NotAvailable)
+    }
```

Comment:
Ideally this would be a compilation time error, where `ColdWalletClient` does NOT implement `NodeInterface` at all.

In the `NodeController<N>`, only the methods that require communication with node would require the `N: NodeInterface` constraint. Methods that do not need to communicate with the node leave `N` unconstrained.

It is probably not a good time to do it now but please file an issue.